loom/script/reflection/lsMemberInfo.h

```diff
@@ -139,12 +140,40 @@ class TemplateInfo {
 public:
     TemplateInfo() : type(NULL)
     {
+        refs = 0;
+    }
+
+    ~TemplateInfo()
+    {
+        lmAssert(refs == 0, "Destructing a template info with %d remaining references", refs);
+    }
+
+    void addReference()
+    {
+        refs++;
+        for (UTsize i = 0; i < types.size(); i++)
+        {
+            types.at(i)->addReference();
+        }
+    }
+    
+    void removeReference()
+    {
+        refs--;
+        for (UTsize i = 0; i < types.size(); i++)
+        {
+            types.at(i)->removeReference();
+        }
+        if (refs <= 0) lmDelete(NULL, this);
```

Comment:
Should probably assert if < 0, since it means we f'ed up the count? This is one of few cases where I like to enforce exactly zero since otherwise you get gnarly lingering corner case issues.